src/client/src/system/service/connection.ts

```diff
@@ -208,46 +156,41 @@ export class Connection {
     payload: TPayload,
     responseTopic: string = ''
   ) {
-    return new Observable<TResult>(obs => {
-      log.debug(
-        `Doing a RPC to [${remoteProcedure}]. Is connected [${
-          this.isConnected
-        }]`
+    return this.connectionStream.pipe(
+      filter(
+        (connection): connection is ConnectionOpenEvent =>
+          connection.type === ConnectionEventType.CONNECTED
+      ),
+      mergeMap(
+        ({ session }) =>
+          new Observable<TResult>(obs => {
+            log.debug(`Doing a RPC to [${remoteProcedure}]]`)
+
+            const dto: SubscriptionRequest<TPayload> = [
+              {
+                payload,
+                replyTo: responseTopic,
+                Username: this.userName
```

Comment:
any reason for the different casing?

Reply:
The server takes the uppercase version, So We convert it at the last possible moment